### PR TITLE
Enhancement/skale 3522 link node addresses

### PR DESCRIPTION
**Pull request by payvint**
Link skale-manager node addresses to IMA

### CODE REVIEW

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by DimaStebaev — DISMISSED** [review]

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by yavrsky — DISMISSED** [review]

File: proxy/contracts/MessageProxyForMainnet.sol [hunk 1]
```diff
@@ -150,30 +148,6 @@ contract MessageProxyForMainnet is Initializable {
...
-     * 
-     * - `msg.sender` must be an owner.
-     */
-    function removeAuthorizedCaller(address caller) external {
-        require(msg.sender == owner, "Sender is not an owner");
-        authorizedCaller[caller] = false;
-    }
-
     /**
      * @dev Adds a `newChainID`.
```
Comment (cstrangedk) [inline proxy/contracts/MessageProxyForMainnet.sol]: Allows LockAndData to add a...

File: proxy/contracts/MessageProxyForMainnet.sol [hunk 2]
```diff
@@ -422,6 +387,20 @@ contract MessageProxyForMainnet is Initializable {
             isValidMessage = true;
     }
 
+    function isAuthorizedCaller(address sender) public view returns (bool) {
```
Comment (cstrangedk) [inline proxy/contracts/MessageProxyForMainnet.sol:393]: missing documentation

**Review by kladkogex — DISMISSED** [review]

File: proxy/contracts/LockAndDataForMainnet.sol [hunk 3]
```diff
@@ -195,6 +192,12 @@ contract LockAndDataForMainnet is OwnableForMainnet {
 
     function initialize() public override initializer {
         OwnableForMainnet.initialize();
-        authorizedCaller[msg.sender] = true;
+    }
+
+    function isSchainOwner(address sender, bytes32 schainId) public virtual view returns (bool) {
```
Comment (cstrangedk) [inline proxy/contracts/LockAndDataForMainnet.sol:200]: add docs

Checks whether sender is owner of SKALE chain.

File: proxy/contracts/MessageProxyForMainnet.sol [hunk 4]
```diff
@@ -215,9 +183,7 @@ contract MessageProxyForMainnet is Initializable {
      * - `msg.sender` must be owner.
      * - `newChainID` must be initialized.
      */
-    function removeConnectedChain(string calldata newChainID) external {
-        require(authorizedCaller[msg.sender], "Not authorized caller");
-
+    function removeConnectedChain(string calldata newChainID) external allow("LockAndData") {
```
Comment (cstrangedk) [inline proxy/contracts/MessageProxyForMainnet.sol:186]: change docs

Allows LockAndData to remove...

**Review by cstrangedk — CHANGES_REQUESTED** [review]

**Review by cstrangedk — DISMISSED** [review]

**Review by sergiy-skalelabs — APPROVED** [review]

**Review by cstrangedk — APPROVED** [review]

**Review by yavrsky — APPROVED** [review]

**Pull request merged**